code-orange/RecordedLocation.swift

```diff
@@ -8,100 +8,58 @@
-  static var iSO8601DateWithMillisec: DateFormatter {
-    let dateFormatter = DateFormatter()
-    dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
-    return dateFormatter
+  private enum CodingKeys: String, CodingKey {
+    case locations
   }
 }
 
-struct ServerRecordedLocation: Equatable, Codable {
-  var lat: CLLocationDegrees
-  var lon: CLLocationDegrees
-  var startTime: String
-  var endTime: String
-  var radius: Int
-  var name: String?
-  var comments: String?
-  
-  init(lat: CLLocationDegrees,
-       lon: CLLocationDegrees,
-       startTime: String,
-       endTime: String,
-       radius: Int,
-       name: String?,
-       comments: String?) {
-    self.lat = lat
-    self.lon = lon
-    self.startTime = startTime
-    self.endTime = endTime
-    self.radius = radius
-    self.name = name
-  }
-}
+struct COLocation: Equatable, Codable {
+  let lat: CLLocationDegrees
+  let lon: CLLocationDegrees
+  let startTime: Date
+  let endTime: Date
+  let radius: Double
+  let name: String
+  let comments: String
 
-struct RecordedLocation: Equatable, Codable {
-  var location: CoronaLocation
-  var startTime: Date
-  var endTime: Date
-  var address: String?
+  private enum CodingKeys: String, CodingKey {
+    case lat
+    case lon
+    case startTime
+    case endTime
+    case radius
+    case name
+    case comments
+  }
 }
 
-extension RecordedLocation {
-  init?(serverLocation: ServerRecordedLocation) {
-    self.location = CoronaLocation(lat: serverLocation.lat, lon: serverLocation.lon)
-    guard let startTime = dateFormatter.date(from: serverLocation.startTime),
-      let endTime = dateFormatter.date(from: serverLocation.endTime) else {
-        return nil
-    }
-    
-    self.startTime = startTime
-    self.endTime = endTime
-    self.address = serverLocation.name
-  }
-  
-  func isLocationColliding(with otherRecordedLocation: RecordedLocation,
+extension COLocation {
+  func isLocationColliding(with otherLocation: COLocation,
                            collisionThresholdInMeters: Double) -> Bool {
-    return self.location.isColliding(with: otherRecordedLocation.location,
-                                    collisionThresholdInMeters: collisionThresholdInMeters)
+    let location1 = CLLocation(latitude: lat, longitude: lon)
+    let location2 = CLLocation(latitude: otherLocation.lat, longitude: otherLocation.lon)
+
+    return location1.distance(from: location2) <= collisionThresholdInMeters
   }
-  
-  func isTimeColliding(with otherRecordedLocation: RecordedLocation, collisionThresholdInSecs: Double) -> Bool {
+
+  func isTimeColliding(with otherRecordedLocation: COLocation,
+                       collisionThresholdInSecs: Double) -> Bool {
     /*
      * we want at least one of the four couples s1-s2, s1-e2, e1-s2, e1-e2
      * to be less than matchingTimeThresholdInSecs apart from each other
      */
-    let s1AndE1 = [self.startTime, self.endTime]
+    let s1AndE1 = [startTime, endTime]
```

Comment:
s1AndE1 is a bit a strange name, consider something more verbose like
span1, span2, etc

Reply:
Agree, although I just move existing code.

Reply:
Your call ☎️ 🙏